Exception occurred in configuring ackTimeout attribute, which caused the project to fail to start.

Configure consumer ackTimeout according to the document.
The project cannot be started.
<img width="847" alt="Image" src="https://github.com/user-attachments/assets/bf9cb305-dd08-4408-b789-d07b00735b99" />

The exception log is as follows：
Caused by: java.lang.IllegalStateException: Error starting listener container [org.springframework.Pulsar.PulsarListenerEndpointContainer#0-0]
	at org.springframework.pulsar.listener.DefaultPulsarMessageListenerContainer.doStart(DefaultPulsarMessageListenerContainer.java:141)
	at org.springframework.pulsar.listener.AbstractPulsarMessageListenerContainer.start(AbstractPulsarMessageListenerContainer.java:103)
	at org.springframework.pulsar.listener.ConcurrentPulsarMessageListenerContainer.doStart(ConcurrentPulsarMessageListenerContainer.java:100)
	at org.springframework.pulsar.listener.AbstractPulsarMessageListenerContainer.start(AbstractPulsarMessageListenerContainer.java:103)
	at org.springframework.pulsar.config.GenericListenerEndpointRegistry.startIfNecessary(GenericListenerEndpointRegistry.java:234)
	at org.springframework.pulsar.config.GenericListenerEndpointRegistry.start(GenericListenerEndpointRegistry.java:186)
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:323)
	... 13 common frames omitted
Caused by: java.lang.RuntimeException: Failed to load config into existing configuration data
	at org.apache.pulsar.client.impl.conf.ConfigurationDataUtils.loadData(ConfigurationDataUtils.java:59)
	at org.apache.pulsar.client.impl.ConsumerBuilderImpl.loadConf(ConsumerBuilderImpl.java:89)
	at org.springframework.pulsar.core.ConsumerBuilderConfigurationUtil.loadConf(ConsumerBuilderConfigurationUtil.java:66)
	at org.springframework.pulsar.listener.DefaultPulsarMessageListenerContainer$Listener.lambda$new$0(DefaultPulsarMessageListenerContainer.java:336)
	at org.springframework.pulsar.core.DefaultPulsarConsumerFactory.lambda$createConsumer$1(DefaultPulsarConsumerFactory.java:119)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.springframework.pulsar.core.DefaultPulsarConsumerFactory.createConsumer(DefaultPulsarConsumerFactory.java:119)
	at org.springframework.pulsar.listener.DefaultPulsarMessageListenerContainer$Listener.<init>(DefaultPulsarMessageListenerContainer.java:343)
	at org.springframework.pulsar.listener.DefaultPulsarMessageListenerContainer.doStart(DefaultPulsarMessageListenerContainer.java:131)
	... 19 common frames omitted
Caused by: org.apache.pulsar.shade.com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "ackTimeout" (class org.apache.pulsar.client.impl.conf.ConsumerConfigurationData), not marked as ignorable (37 known properties: "priorityLevel", "autoUpdatePartitions", "acknowledgementsGroupTimeMicros", "receiverQueueSize", "regexSubscriptionMode", "subscriptionName", "retryEnable", "topicsPattern", "autoAckOldestChunkedMessageOnQueueFull", "maxPendingChunkedMessage", "batchIndexAckEnabled", "ackReceiptEnabled", "maxAcknowledgmentGroupSize", "topicNames", "cryptoFailureAction", "subscriptionMode", "consumerName", "expireTimeOfIncompleteChunkedMessageMillis", "maxTotalReceiverQueueSizeAcrossPartitions", "subscriptionProperties", "patternAutoDiscoveryPeriod", "properties", "subscriptionType", "autoUpdatePartitionsIntervalSeconds", "deadLetterPolicy", "ackTimeoutMillis", "maxPendingChuckedMessage", "autoScaledReceiverQueueSizeEnabled", "poolMessages", "replicateSubscriptionState", "readCompacted", "resetIncludeHead", "subscriptionInitialPosition", "negativeAckRedeliveryDelayMicros", "tickDurationMillis", "startPaused", "topicConfigurations"])
 at [Source: (String)"{"autoUpdatePartitionsIntervalSeconds":60,"ackTimeout":"60s","tickDurationMillis":1000,"replicateSubscriptionState":false,"negativeAckRedeliveryDelayMicros":60000000,"expireTimeOfIncompleteChunkedMessageMillis":60000,"maxPendingChuckedMessage":10,"topicNames":["topic-1"],"priorityLevel":0,"ackReceiptEnabled":false,"acknowledgementsGroupTimeMicros":100000,"receiverQueueSize":1000,"autoUpdatePartitions":true,"subscriptionMode":"Durable","poolMessages":false,"patternAutoDiscoveryPeriod":60,"subscri"[truncated 549 chars]; line: 1, column: 57] (through reference chain: org.apache.pulsar.client.impl.conf.ConsumerConfigurationData["ackTimeout"])
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1132)
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:2202)
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1705)
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1683)
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:320)
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4730)
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3677)
	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3645)
	at org.apache.pulsar.client.impl.conf.ConfigurationDataUtils.loadData(ConfigurationDataUtils.java:57)

--- a/spring-pulsar-docs/src/main/antora/modules/ROOT/pages/reference/pulsar/message-consumption.adoc
+++ b/spring-pulsar-docs/src/main/antora/modules/ROOT/pages/reference/pulsar/message-consumption.adoc
@@ -625,12 +625,12 @@ Apache Pulsar provides various native strategies for message redelivery and erro
 By default, Pulsar consumers do not redeliver messages unless the consumer crashes, but you can change this behavior by setting an ack timeout on the Pulsar consumer.
 If the ack timeout property has a value above zero and if the Pulsar consumer does not acknowledge a message within that timeout period, the message is redelivered.
 
-When you use Spring for Apache Pulsar, you can set this property via a <<_consumer_customization_on_pulsarlistener,consumer customizer>> or with the native Pulsar `ackTimeout` property in the `properties` attribute of `@PulsarListener`:
+When you use Spring for Apache Pulsar, you can set this property via a <<_consumer_customization_on_pulsarlistener,consumer customizer>> or with the native Pulsar `ackTimeoutMillis` property in the `properties` attribute of `@PulsarListener`:
 
 [source, java]
 ----
 @PulsarListener(subscriptionName = "subscription-1", topics = "topic-1"
-                properties = {"ackTimeout=60s"})
+                properties = {"ackTimeoutMillis=60000"})
 public void listen(String s) {
     ...
 }
@@ -649,7 +649,7 @@ class AckTimeoutRedeliveryConfig {
     @PulsarListener(subscriptionName = "withAckTimeoutRedeliveryBackoffSubscription",
             topics = "withAckTimeoutRedeliveryBackoff-test-topic",
             ackTimeoutRedeliveryBackoff = "ackTimeoutRedeliveryBackoff",
-            properties = { "ackTimeout=60s" })
+            properties = { "ackTimeoutMillis=60000" })
     void listen(String msg) {
         // some long-running process that may cause an ack timeout
     }
@@ -720,7 +720,7 @@ class DeadLetterPolicyConfig {
 
     @PulsarListener(id = "deadLetterPolicyListener", subscriptionName = "deadLetterPolicySubscription",
             topics = "topic-with-dlp", deadLetterPolicy = "deadLetterPolicy",
-            subscriptionType = SubscriptionType.Shared, properties = { "ackTimeout=1s" })
+            subscriptionType = SubscriptionType.Shared, properties = { "ackTimeoutMillis=1000" })
     void listen(String msg) {
         throw new RuntimeException("fail " + msg);
     }
@@ -743,7 +743,7 @@ This bean specifies a number of things, such as the max delivery (10, in this ca
 If you do not specify a DLQ topic name, it defaults to `<topicname>-<subscriptionname>-DLQ` in Pulsar.
 Next, we provide this bean name to `PulsarListener` by setting the `deadLetterPolicy` property.
 Note that the `PulsarListener` has a subscription type of `Shared`, as the DLQ feature only works with shared subscriptions.
-This code is primarily for demonstration purposes, so we provide an `ackTimeout` value of 1 second.
+This code is primarily for demonstration purposes, so we provide an `ackTimeoutMillis` value of 1000.
 The idea is that the code throws the exception and, if Pulsar does not receive an ack within 1 second, it does a retry.
 If that cycle continues ten times (as that is our max redelivery count in the `DeadLetterPolicy`), the Pulsar consumer publishes the messages to the DLQ topic.
 We have another `PulsarListener` that listens on the DLQ topic to receive data as it is published to the DLQ topic.

--- a/spring-pulsar-docs/src/main/antora/modules/ROOT/pages/reference/reactive-pulsar/reactive-message-consumption.adoc
+++ b/spring-pulsar-docs/src/main/antora/modules/ROOT/pages/reference/reactive-pulsar/reactive-message-consumption.adoc
@@ -88,6 +88,7 @@ Flux<MessageResult<Void>> listen(Flux<org.apache.pulsar.client.api.Message<Strin
     return messages
         .doOnNext((msg) -> System.out.println("Received: " + msg.getValue()))
         .map(MessageResult::acknowledge);
+}
 ----
 Here we receive the records as a `Flux` of Pulsar messages.
 In addition, to enable stream consumption at the `ReactivePulsarListener` level, you need to set the `stream` property on the annotation to `true`.
@@ -293,7 +294,7 @@ You can specify this property directly as a Pulsar consumer property via a <<rea
 ----
 @Bean
 ReactiveMessageConsumerBuilderCustomizer<String> consumerCustomizer() {
-    return b -> b.property("ackTimeout", "60s");
+    return b -> b.property("ackTimeoutMillis", "60000");
 }
 ----
 
@@ -342,7 +343,7 @@ class DeadLetterPolicyConfig {
 
     @Bean
     ReactiveMessageConsumerBuilderCustomizer<String> ackTimeoutCustomizer() {
-        return b -> b.property("ackTimeout", "1s");
+        return b -> b.property("ackTimeoutMillis", "1000");
     }
 }
 ----
@@ -352,7 +353,7 @@ This bean specifies a number of things, such as the max delivery (10, in this ca
 If you do not specify a DLQ topic name, it defaults to `<topicname>-<subscriptionname>-DLQ` in Pulsar.
 Next, we provide this bean name to `ReactivePulsarListener` by setting the `deadLetterPolicy` property.
 Note that the `ReactivePulsarListener` has a subscription type of `Shared`, as the DLQ feature only works with shared subscriptions.
-This code is primarily for demonstration purposes, so we provide an `ackTimeout` value of 1 second.
+This code is primarily for demonstration purposes, so we provide an `ackTimeoutMillis` value of 1000.
 The idea is that the code throws the exception and, if Pulsar does not receive an ack within 1 second, it does a retry.
 If that cycle continues ten times (as that is our max redelivery count in the `DeadLetterPolicy`), the Pulsar consumer publishes the messages to the DLQ topic.
 We have another `ReactivePulsarListener` that listens on the DLQ topic to receive data as it is published to the DLQ topic.

